biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -855,7 +855,7 @@ private void doSecurity() {
 	}
 
 	private boolean isFragment(Bundle b) {
-		return padmin != null && padmin.getBundleType(b) == PackageAdmin.BUNDLE_TYPE_FRAGMENT;
+		return b.getHeaders().get(Constants.FRAGMENT_HOST) != null;
```

Comment:
PackageAdmin is the right way to do things if present. You could change the code to check the headers IFF padmin is null.